eclair-core/src/test/scala/fr/acinq/eclair/router/RouteCalculationSpec.scala

```diff
@@ -1762,18 +1762,69 @@ class RouteCalculationSpec extends AnyFunSuite with ParallelTestExecution {
       assert(route2Ids(route) === 0 :: 2 :: 5 :: 6 :: 7 :: 4 :: Nil)
     }
     { // small base hop cost
-      val Success(routes) = findRoute(g, start, b, DEFAULT_AMOUNT_MSAT, 100000000 msat, numRoutes = 1, routeParams = DEFAULT_ROUTE_PARAMS.copy(ratios = WeightRatios(1, 0, 0, 0, RelayFees(100 msat, 0))), currentBlockHeight = 400000)
+      val Success(routes) = findRoute(g, start, b, DEFAULT_AMOUNT_MSAT, 100000000 msat, numRoutes = 1, routeParams = DEFAULT_ROUTE_PARAMS.copy(ratios = Left(WeightRatios(1, 0, 0, 0, RelayFees(100 msat, 0)))), currentBlockHeight = 400000)
       assert(routes.distinct.length == 1)
       val route :: Nil = routes
       assert(route2Ids(route) === 0 :: 2 :: 3 :: 4 :: Nil)
     }
     { // large proportional hop cost
-      val Success(routes) = findRoute(g, start, b, DEFAULT_AMOUNT_MSAT, 100000000 msat, numRoutes = 1, routeParams = DEFAULT_ROUTE_PARAMS.copy(ratios = WeightRatios(1, 0, 0, 0, RelayFees(0 msat, 200))), currentBlockHeight = 400000)
+      val Success(routes) = findRoute(g, start, b, DEFAULT_AMOUNT_MSAT, 100000000 msat, numRoutes = 1, routeParams = DEFAULT_ROUTE_PARAMS.copy(ratios = Left(WeightRatios(1, 0, 0, 0, RelayFees(0 msat, 200)))), currentBlockHeight = 400000)
       assert(routes.distinct.length == 1)
       val route :: Nil = routes
       assert(route2Ids(route) === 0 :: 1 :: Nil)
     }
   }
+
+  test("most likely successful path") {
+    //  S ---> A ---> B
+    //         |      ^
+    //         v      |
+    //         C ---> D
+    val start = randomKey().publicKey
+    val g = DirectedGraph(List(
+      makeEdge(0L, start, a, 0 msat, 0),
+      makeEdge(1L, a, b, 1000 msat, 1000, capacity = (DEFAULT_AMOUNT_MSAT * 1.2).truncateToSatoshi),
+      makeEdge(2L, a, c, 400 msat, 500, capacity = (DEFAULT_AMOUNT_MSAT * 3).truncateToSatoshi),
+      makeEdge(3L, c, d, 400 msat, 500, capacity = (DEFAULT_AMOUNT_MSAT * 3).truncateToSatoshi),
+      makeEdge(4L, d, b, 400 msat, 500, capacity = (DEFAULT_AMOUNT_MSAT * 3).truncateToSatoshi),
+    ))
+
+    val hc = HeuristicsConstants(
+      lockedFundsRisk = 0.0,
+      failureCost = RelayFees(1000 msat, 500),
+      hopCost = RelayFees(0 msat, 0),
+    )
```

Comment:
It feels to me that this configuration is essentially the same as the normal weight ratios with a big factor for the channel capacity component, isn't it? But maybe it's mostly because for now we use a uniform distribution for the likely balance of the channel...

Reply:
The goal is the same indeed but the result should be very different. With the current capacity ratio, a channel of 1BTC gets a score five times bigger than a channel of 0.2BTC, but when you want to pay 10k sats, both capacities are vastly larger than the payment amount so they should both get the maximum score (for the capacity part).